### PR TITLE
Task/jenkins 40256 enhance form controls

### PR DESCRIPTION
**Pull request by cliffmeyers**
**Description**
- See [JENKINS-40256](https://issues.jenkins-ci.org/browse/JENKINS-40256).

**Submitter checklist**
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests

**Reviewer checklist**
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by sophistifunk** [comment]
Can we get some screenshots? My checkout is kinda wack ATM so I don't wanna stash it and switch :D

**Comment by cliffmeyers** [comment]
Some screenshots:

style-guide-1-buttons

style-guide-2-checkbox

style-guide-3-radiobuttongroup

style-guide-4-textinput

style-guide-5-textinput-sizes

style-guide-6-textarea

style-guide-7-textarea-sizes

style-guide-8-dropdown

**Comment by sophistifunk** [comment]
Screenshots look sweeeeeet

**Comment by i386** [comment]
nice work @cliffmeyers and @brody 👍

**Comment by brody** [comment]
@cliffmeyers Looking good! Just a couple of tweaks.

Also, it might just be the screenshots, but the inner shadow looks a little harsh. Is it `box-shadow: inset 0 0 2px 3px rgba(0,0,0,0.20);`?

fine tuning

**Comment by cliffmeyers** [comment]
@brody I'll make those small tweaks to text controls and radio button. Regarding the box shadow, I actually had to use this:

```
// need to use blur-radius=1px instead of 2px to avoid IE/Edge centering issue
box-shadow: inset 0 0 1px 3px rgba(0, 0, 0, 0.2);
```

It looked almost indistinguishable to me on a non-retina monitor but maybe it's not too clean on HDPI. In retrospect perhaps I can use some conditional CSS to tweak it for MS browsers.

I made the "selected" option bold because I wanted a way to distinguish it from others. If you have another idea I am all ears :)

**Comment by cliffmeyers** [comment]
@brody to clarify, are the text controls supposed to have only a blue border rect on focus, or a blue border rect in additional the box-shadow? I would really recommend the latter since border color change alone is a little too subtle IMO, but I'll impl it however you think is best.

**Comment by cliffmeyers** [comment]
@sophistifunk worked around some annoying visual issues in IE11 and Edge in this commit:  4d2c7db  could not find a better way, but if you have ideas I'm all ears.

**Comment by brody** [comment]
@cliffmeyers 
box-shadow // damn you MS. I was able to pick the 1px (100%) difference by eye, so it should soften the shadow a fair bit.
dropdown // i figured the label at the top showed which was selected and the blue was used for hover.
focus // text controls just have the blue border on focus, the box-shadow would get in the way of the text being inputted.

File: less/components/button.less [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+// Apply the mixin to the buttons
+button, .btn, .btn-default, .btn-primary {
+    // TODO: need to derive hover and active from @brand-primary
+    .btn-styles(
+        @btn-primary-bg,
+        @btn-primary-border,
+        @btn-primary-color,
+        #3080de,
```
Comment (kzantow) [inline less/components/button.less]: should these be moved to variables?

File: less/theme.less [hunk 2]
```diff
@@ -196,96 +192,15 @@ body {
...
-    &:active,
-    &.active {
-        background-color: @btn-color;
-        border-color: darken(@btn-border, 12%);
-        color: @btn-text;
-        text-decoration: none;
+.browser-ie11({
+    // fix egregious vertical centering issue with Lato in IE11 by falling back to sans-serif
+    body {
+        font-family: sans-serif !important;
```
Comment (kzantow) [inline less/theme.less]: ehhh... you could maybe just make this more specific, if there was a class on `body`, e.g. `body.ie11.ie11` or whatever

**Comment by scherler** [comment]
:bee: looks nice

**Comment by sophistifunk** [comment]
bbm-bueno

**Comment by cliffmeyers** [comment]
I will merge this as soon as I have a chance to test the changes in `blueocean-plugin`. As there are a lot of style changes, this may take a little while before I'm able to wrap it up.

**Pull request merged**